packages/knip/src/typescript/resolveModuleNames.ts

```diff
@@ -35,6 +34,32 @@ export function createCustomModuleResolver(
   const customCompilerExtensionsSet = new Set(customCompilerExtensions);
   const extensions = [...DEFAULT_EXTENSIONS, ...customCompilerExtensions];
 
+  const virtualDeclarationFiles = new Map<string, { path: string; ext: string }>();
+
+  const tsSys: ts.System = {
+    ...ts.sys,
+    // We trick TypeScript into resolving paths with arbitrary extensions. When
+    // a module "./foo.ext" is imported TypeScript only tries to resolve it to
+    // "./foo.d.ext.ts". TypeScript never checks whether "./foo.ext" itself exists.
```

Comment:
Don't you mean `module.ext.ts` and `module.ext.d.ts` (and same for `.mts` + `.cts`)?

Reply:
I was confused by this as well but it is correct. If the import has an unknown extension like `.../foo.ext` typescript looks for `.../foo.d.ext.ts` ([code in TypeScript](https://github.com/microsoft/TypeScript/blob/46410044add2e9f53cea58e445de18dcda53443f/src/compiler/moduleNameResolver.ts#L2161)).

Reply:
TIL! Thanks for figuring that out + the link, much appreciated.